### DISCUSSION TITLE
In windows, I wrote a class to draw a rectangle, when I maximize the window, I found the bug.

### DISCUSSION THREAD

**Discussion by xuyun018** [fltk/fltk#1027] [Q&A]
**Expected behavior**

**Screenshots**

**FLTK Version**
 - Version: [e.g. 1.4.0]

**Operating System / Platform:**
Please be as precise as possible, e.g. "Linux: Ubuntu 20.04"
 - OS: Windows
 - OS Version: Windows 7, Windows 10
 - Processor if relevant [Intel]

The image at the bottom of the window cannot be updated, and the screen size is 1366 * 768

My code:
main.cpp
...

Fl_Window* window = new Fl_Double_Window(1024, 600);
    window->color(FL_WHITE);
    MyWidget* wgt = new MyWidget(8, 8, 800 - 16, 600 - 16);
    window->end();
    window->resizable(window);
    window->show(argc, argv);

    Fl::run();
...
//---------------------------------------------------------------------------
mywidget.h
//---------------------------------------------------------------------------
#ifndef MYWIDGET_H
#define MYWIDGET_H
//---------------------------------------------------------------------------
#include <Fl/Fl.h>
#include <Fl/Fl_Widget.h>
#include <Fl/Fl_Image.h>
//---------------------------------------------------------------------------
class MyWidget : public Fl_Widget {
public:
	MyWidget(int x, int y, int width, int height);
	virtual ~MyWidget();

	void rotate();

	void draw() FL_OVERRIDE;
	void resize(int x, int y, int width, int height) FL_OVERRIDE;

private:
	int generate_image(int width, int height);

	unsigned char i_;

	int ox_, oy_;
	int width_, height_;

	int left_, top_;
	unsigned char *pdata_;
	unsigned int datasize_;
};
//---------------------------------------------------------------------------
void wgt_on_timer(void *data);
//---------------------------------------------------------------------------
#endif
//---------------------------------------------------------------------------
mywidget.cpp:
//---------------------------------------------------------------------------
#include "mywidget.h"
//---------------------------------------------------------------------------
void wgt_on_timer(void *data)
{
	MyWidget *wgt = (MyWidget *)data;

	wgt->rotate();

	Fl::repeat_timeout(0.1f, wgt_on_timer, data);
}

MyWidget::MyWidget(int x, int y, int width, int height)
: Fl_Widget(x, y, width, height)
{
	i_ = 0;

	ox_ = 0;
	oy_ = 0;
	width_ = 0;
	height_ = 0;

	left_ = x;
	top_ = y;

	pdata_ = NULL;
	datasize_ = 0;

	generate_image(width, height);

	Fl::add_timeout(0.1f, wgt_on_timer, this);
}
MyWidget::~MyWidget()
{
	Fl::remove_timeout(wgt_on_timer, this);

	if (pdata_)
	{
		delete[] pdata_;
		pdata_ = NULL;
	}
	datasize_ = 0;
}

void MyWidget::rotate()
{
	i_++;
	if (visible())
	{
		redraw();
	}
}

int MyWidget::generate_image(int width, int height)
{
	int result = 0;
	int wh = 0;

	if (width > 90 && height > 90)
	{
		wh = width < height ? width : height;
	}
	if (wh > 90)
	{
		if (wh != width_ || wh != height_)
		{
			width_ = wh;
			height_ = wh;

			ox_ = (width - wh) >> 1;
			oy_ = 8;

			if (pdata_)
			{
				delete[] pdata_;
			}
			datasize_ = wh * wh * 3;
			pdata_ = new unsigned char[datasize_];

			result = 1;
		}
	}
	else
	{
		width_ = 0;
		height_ = 0;
	}

	return(result);
}

void MyWidget::draw()
{
	unsigned int width, height;

	width = width_;
	height = height_;

	if (width == 0 || height == 0)
	{
		return;
	}

	memset(pdata_, i_, datasize_);

	Fl_RGB_Image* pimage;

	pimage = new Fl_RGB_Image(pdata_, width, height, 3);
	pimage->draw(left_, top_);
	delete pimage;
}

void MyWidget::resize(int x, int y, int width, int height)
{
	left_ = x;
	top_ = y;
	generate_image(width, height);
}
//---------------------------------------------------------------------------

**Comment by xuyun018** [comment]
Ubuntu 22.04 desktop amd64, has this bug too.

**Comment by ManoloFLTK** [comment]
Your error is in `MyWidget::resize(int x, int y, int width, int height)`. When you override such a function, it is very important to call the corresponding function of the parent class which has things to do. I attach a corrected version of your program:

[xuyun.cxx.txt](https://github.com/user-attachments/files/16415508/xuyun.cxx.txt)

**Reply by xuyun018** [reply]
It's OK! My fault.

**Comment by Albrecht-S** [comment]
@xuyun018 You posted a lot of code but it's not clear to me if all your fragments constitute a full, compileable program. If you need help with your program, please post a full, compileable (and linkable) program in one source file. You can upload the entire (`.cxx` or `.cpp`) file if you rename it to `.txt` (or append `.txt` to your original filename). If you really need a header file, please post this as well, or include both files in a `.zip` file and post that file.

Please try to reduce your example program to the bare minimum that shows the described behavior, otherwise we can't help efficiently. This test program would ideally be no longer than about 100 lines.

> The image at the bottom of the window cannot be updated, and the screen size is 1366 * 768

This description is not sufficient (for me). Please elaborate what you mean with "cannot be updated", and why "the screen size is 1366 * 768" is wrong. Please tell us what you expect instead.

I'm sorry if I misunderstood your report but it's too complicated to put all things together and try to find out. I moved your issue to a discussion (Q&A = Questions and Answers) and I hope we can help you if you provide a short test program. Thanks for your understanding.

**Note:** Meanwhile I saw that Manolo found an answer, but I'm leaving my post here anyway for further reference.

**Reply by xuyun018** [reply]
ManoloFLTK solved my bug. Not FLTK's bug.
